services/flows-operator/src/FlowOperator.js

```diff
@@ -313,6 +340,81 @@ class FlowOperator {
         }, {});
         return Object.assign(envVars, node.env);
     }
+
+    async _createFlowSecret(flow) {
+        this._logger.debug('About to create a secret');
+        const credentials = await this._createAmqpCredentials(flow);
+
+        const flowSecret = new FlowSecret({
+            metadata: {
+                name: flow.id,
+                namespace: this._config.get('NAMESPACE')
+            },
+            data: {
+                AMQP_URI: this._prepareAmqpUri(credentials)
+            }
+        });
+
+        await this._coreClient.secrets.post({
+            body: flowSecret.toDescriptor()
+        });
+
+        this._logger.debug('Secret has been created');
+    }
+
+    async _deleteFlowSecret(flow) {
+        this._logger.debug('About to delete a secret');
+        const secret = await this._getFlowSecret(flow);
+        if (!secret) {
+            return;
+        }
+
+        const username = secret.amqpUsername;
+        await this._deleteAmqpCredentials({ username });
+        this._logger.trace('Removed AMQP credentials');
+
+        await this._coreClient.secrets(secret.id).delete();
+        this._logger.debug('Secret has been deleted');
+    }
+
+    async _getFlowSecret(flow) {
+        try {
+            const result = await this._coreClient.secrets(flow.id).get();
+            return FlowSecret.fromDescriptor(result.body);
+        } catch (e) {
+            return null;
+        }
+    }
+
+    _prepareAmqpUri({ username, password }) {
+        const baseUri = new URL(this._config.get('RABBITMQ_URI_FLOWS'));
+        baseUri.username = username;
+        baseUri.password = password;
+
+        return baseUri.toString();
+    }
+
+    async _createAmqpCredentials(flow) {
+        const username = uuid.v4(); //@todo: use flow.id as a username?
```

Comment:
I think it is good idea: it would be easier to debug what's happening in case of problem, as it makes easier to find out who is initiator of connection to rabbit